Incompatibilities with boost-1.64

Hi all.

I'm rebuilding `OpenMS-2.2.0` against `Boost-1.64` for `Fedora` devel branch but it fails with following error:

```
In file included from /usr/include/boost/numeric/ublas/vector.hpp:21:0,
                 from /usr/include/boost/numeric/ublas/matrix.hpp:18,
                 from /usr/include/boost/accumulators/statistics/covariance.hpp:22,
                 from /builddir/build/BUILD/openms-2.2.0/py2build/src/openms/include/OpenMS/MATH/STATISTICS/StatisticFunctions.h:42,
                 from /builddir/build/BUILD/openms-2.2.0/py2build/src/openms/source/MATH/STATISTICS/LinearRegression.cpp:37:
/usr/include/boost/numeric/ublas/storage.hpp: In member function 'void boost::numeric::ublas::unbounded_array<T, ALLOC>::serialize(Archive&, unsigned int)':
/usr/include/boost/numeric/ublas/storage.hpp:299:33: error: 'make_array' is not a member of 'boost::serialization'
             ar & serialization::make_array(data_, s);
                                 ^~~~~~~~~~
/usr/include/boost/numeric/ublas/storage.hpp:299:33: note: suggested alternative: 'make_nvp'
             ar & serialization::make_array(data_, s);
                                 ^~~~~~~~~~
                                 make_nvp
/usr/include/boost/numeric/ublas/storage.hpp: In member function 'void boost::numeric::ublas::bounded_array<T, N, ALLOC>::serialize(Archive&, unsigned int)':
/usr/include/boost/numeric/ublas/storage.hpp:494:33: error: 'make_array' is not a member of 'boost::serialization'
             ar & serialization::make_array(data_, s);
                                 ^~~~~~~~~~
/usr/include/boost/numeric/ublas/storage.hpp:494:33: note: suggested alternative: 'make_nvp'
             ar & serialization::make_array(data_, s);
                                 ^~~~~~~~~~
                                 make_nvp
In file included from /usr/include/boost/accumulators/statistics/covariance.hpp:22:0,
                 from /builddir/build/BUILD/openms-2.2.0/py2build/src/openms/include/OpenMS/MATH/STATISTICS/StatisticFunctions.h:42,
                 from /builddir/build/BUILD/openms-2.2.0/py2build/src/openms/source/MATH/STATISTICS/LinearRegression.cpp:37:
/usr/include/boost/numeric/ublas/matrix.hpp: In member function 'void boost::numeric::ublas::c_matrix<T, M, N>::serialize(Archive&, unsigned int)':
/usr/include/boost/numeric/ublas/matrix.hpp:5977:33: error: 'make_array' is not a member of 'boost::serialization'
             ar & serialization::make_array(data_, N);
                                 ^~~~~~~~~~
/usr/include/boost/numeric/ublas/matrix.hpp:5977:33: note: suggested alternative: 'make_nvp'
             ar & serialization::make_array(data_, N);
                                 ^~~~~~~~~~
                                 make_nvp
make[3]: *** [src/openms/CMakeFiles/OpenMS.dir/build.make:873: src/openms/CMakeFiles/OpenMS.dir/source/MATH/STATISTICS/LinearRegression.cpp.o] Error 1
```

No errors against `boost-1.63`.
I suspect there are some incompatibilities with the newer version of `boost`.

Full build log: https://kojipkgs.fedoraproject.org//work/tasks/9431/20689431/build.log

--- a/src/openms/include/OpenMS/MATH/STATISTICS/StatisticFunctions.h
+++ b/src/openms/include/OpenMS/MATH/STATISTICS/StatisticFunctions.h
@@ -47,7 +47,9 @@
 #include <boost/function/function_base.hpp>
 #include <boost/lambda/casts.hpp>
 #include <boost/lambda/lambda.hpp>
+#if OPENMS_BOOST_VERSION_MINOR > 64
 #include <boost/serialization/array_wrapper.hpp>
+#endif
 
 #include <iterator>
 #include <algorithm>

--- a/src/openms/include/OpenMS/MATH/STATISTICS/StatisticFunctions.h
+++ b/src/openms/include/OpenMS/MATH/STATISTICS/StatisticFunctions.h
@@ -47,6 +47,7 @@
 #include <boost/function/function_base.hpp>
 #include <boost/lambda/casts.hpp>
 #include <boost/lambda/lambda.hpp>
+#include <boost/serialization/array_wrapper.hpp>
 
 #include <iterator>
 #include <algorithm>

